app/SignUp/page.js

```diff
@@ -0,0 +1,153 @@
+"use client";
+import { useRef } from "react";
+import { useState } from "react";
+import { Box, TextField, Button, Typography, Input } from "@mui/material";
+import { Label } from "@mui/icons-material";
+import { useRouter } from "next/navigation";
+
+export default function SignUp() {
+  const router = useRouter();
+
+  const [emailError, setEmailError] = useState(false);
+  const [emailErrorMessage, setEmailErrorMessage] = useState("");
+  const [passwordError, setPasswordError] = useState(false);
+  const [passwordErrorMessage, setPasswordErrorMessage] = useState("");
+  const [firstNameError, setFirstNameError] = useState(false);
```

Comment:
Instead of doing this I'd recommend using one state for the whole form as a JS object like we saw in class.

Reply:
I change it and used the single state object for whole form